neo/rawio/medrawio.py

```diff
@@ -0,0 +1,238 @@
+
+        # Create Neo header structure
+        self.header = {}
+        self.header['nb_block'] = 1
+        self.header['nb_segment'] = [1]  # assume continuous MED session
+        self.header['signal_streams'] = signal_streams
+        self.header['signal_channels'] = signal_channels
+        self.header['spike_channels'] = spike_channels
+        self.header['event_channels'] = event_channels
+
+        # `_generate_minimal_annotations()` must be called to generate the nested
+        # dict of annotations/array_annotations
+        self._generate_minimal_annotations()
+        
+        # Add custom annotations for neo objects
+        bl_ann = self.raw_annotations['blocks'][0]
+        bl_ann['name'] = 'MED Data Block'
+        bl_ann.update(self.sess.session_info)
+        seg_ann = bl_ann['segments'][0]
+        seg_ann['name'] = 'Seg #0 Block #0'
+        
+        # this pprint lines really help for understand the nested (and complicated sometimes) dict
+        #from pprint import pprint
+        #pprint(self.raw_annotations)
+
+    def _segment_t_start(self, block_index, seg_index):
+        return self.session_start_time
+
+    def _segment_t_stop(self, block_index, seg_index):
+        return self.session_end_time
+
+    def _get_signal_size(self, block_index, seg_index, stream_index):
+        return self.num_samples_in_session
+
+    def _get_signal_t_start(self, block_index, seg_index, stream_index):
+        return self.session_start_time
+
+    def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop,
+                                stream_index, channel_indexes):
+        
+        import dhn_med_py
+        from dhn_med_py import MedSession
+        
+        # Correct for None start/stop inputs
+        if i_start is None:
+            i_start = 0
+        if i_stop is None:
+            i_stop = self.num_samples_in_session
+            
+        # Check for invalid start/stop inputs
+        if i_start < 0 or i_stop > self.num_samples_in_session:
+            raise IndexError("MED read error: Too many samples requested!")
+        if i_start > i_stop:
+            raise IndexError("MED read error: i_start (" + i_start + ") is greater than i_stop (" + i_stop + ")")
+
+        stream_id = self.header['signal_streams'][stream_index]['id']
+        
+        num_channels = 0
+        if channel_indexes is None:
+            self.sess.set_channel_active('all')
+            num_channels = self.num_channels_in_session
+        else:
+            if any(channel_indexes < 0):
+                raise IndexError('No channels selected!')
+            # Set all channels to be inactive, then selectively set some of them to be active
+            self.sess.set_channel_inactive('all')
+            for i, channel_idx in enumerate(channel_indexes):
+                num_channels += 1
+                self.sess.set_channel_active(str(self.header['signal_channels'][channel_idx][0]))
```

Comment:
I think the channel name should be already a string, so you probably don't need to convert again here. Also to improve readability maybe write it like this:
```suggestion
                self.sess.set_channel_active(self.header['signal_channels']['channel_name'][channel_idx])
```